core/src/main/java/com/datastrato/gravitino/catalog/CapabilityHelpers.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.catalog;
+
+import static com.datastrato.gravitino.rel.Column.DEFAULT_VALUE_NOT_SET;
+
+import com.datastrato.gravitino.connector.capability.Capability;
+import com.datastrato.gravitino.rel.Column;
+import com.datastrato.gravitino.rel.TableChange;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+
+public class CapabilityHelpers {
+
+  public static Column[] applyCapabilities(Column[] columns, Capability capabilities) {
+    return Arrays.stream(columns)
+        .map(c -> applyCapabilities(c, capabilities))
+        .toArray(Column[]::new);
+  }
+
+  public static TableChange[] applyCapabilities(Capability capabilities, TableChange... changes) {
+    return Arrays.stream(changes)
+        .map(
+            change -> {
+              if (change instanceof TableChange.AddColumn) {
+                return applyCapabilities((TableChange.AddColumn) change, capabilities);
+
+              } else if (change instanceof TableChange.UpdateColumnNullability) {
+                return applyCapabilities(
+                    (TableChange.UpdateColumnNullability) change, capabilities);
+              }
+              return change;
+            })
+        .toArray(TableChange[]::new);
+  }
+
+  private static TableChange applyCapabilities(
+      TableChange.AddColumn addColumn, Capability capabilities) {
+    Column appliedColumn =
+        applyCapabilities(
+            Column.of(
+                addColumn.fieldName()[0],
```

Comment:
Why getting the first element of the `fieldName`?

Reply:
because the first name is the column name, and the others are just nested fields in columns.